src/mlpack/methods/ann/rnn_impl.hpp

```diff
@@ -121,7 +119,7 @@ void RNN<OutputLayerType, InitializationRuleType>::ResetCells()
 {
   for (size_t i = 1; i < network.size(); ++i)
```

Comment:
Should this be from `i = 0` to `network.size()`?  This seems to fail if you have an LSTM as the first layer of a network.

Reply:
You are right, we have to perform the backward pass (layer = 0) if the layer wraps another layer or parameter (in case of the LSTM class). However, it's unnecessary for some layers e.g. linear layer, where the error from the previous layer is used; in this case, we can avoid the extra backward function since there is no layer which might use the calculated error. I used an Identity layer as the first layer to solve the problem, but it's definitely not a good solution.